torch/onnx/symbolic_opset9.py

```diff
@@ -1173,14 +1177,31 @@ def le(g, input, other):
     return gt_impl(g, input, other)
 
 
-@wrap_logical_op_with_cast_to_and_from("Bool")
 def __and_(g, input, other):
-    return g.op("And", input, other)
+    if input.type().scalarType() == "Bool" and \
+            other.type().scalarType() is not None and other.type().scalarType() == "Bool":
```

Comment:
nit: why need both `other.type().scalarType() is not None` and `other.type().scalarType() == "Bool"`?